pghoard/walreceiver.py

```diff
@@ -79,10 +79,16 @@ def fetch_timeline_history_files(self, max_timeline):
             self.c.execute("TIMELINE_HISTORY {}".format(max_timeline))
             timeline_history = self.c.fetchone()
             history_filename = timeline_history[0]
-            history_data = timeline_history[1].tobytes()
+            # Type changed from BYTEA to TEXT with PG14, see
+            # https://git.postgresql.org/gitweb/?p=postgresql.git;a=commit;h=66a8f090485e3e897a4804121fdbe856cba72d70
+            if isinstance(timeline_history[1], memoryview):
+                history_data = timeline_history[1].tobytes()  # pragma: no cover
+            else:
+                history_data = timeline_history[1].encode()  # pragma: no cover
```

Comment:
Why do we need the `pragma: no cover` here ?

Reply:
Depending on the Postgres version either one or the other line is executed. So there's always one line shown as not being covered - although it's actually covered when run with the correct PG version. I think this is confusing and I'd rather not read the comment every time to see why it's shown as not being covered, but I'm happy to remove `pragma: no cover` if we're OK with that.

Reply:
Do we have a coverage report bug where we compute the result only for a single pg version instead of computing the union of the coverage with each versions ?

Reply:
Stupid question: does this thing work better if using a ternary operator like

```history_data = ...tobytes if isinstance ... else ..encode()``` ?

Reply:
> Stupid question: does this thing work better if using a ternary operator like
> 
> `history_data = ...tobytes if isinstance ... else ..encode()` ?

This will replace non-covered lines with a partially covered line.

Reply:
@kmichel-aiven Don't know how it's supposed to work, I was not expecting that there is a step the merges coverage results.

@alanfranz-at-work Hah, I'm sure that works, good idea!

Reply:
:-) So leave the `if`/`else` and remove the `pragma` for now?

Reply:
OK, wasn't aware that results should actually be merged. Let me just remove the `pragma` then.

Reply:
Ok, I took a look at how coverage is computed locally and how tests are run, now I understand the issue :)
For Github Actions and Codecov, the report is automatically merged, but locally testing each pg version is very manual.

I'm trying something to remove your issue of partial coverage on local environment: #524.
This should also speed up the entire test suite.